src/ui/ButtonLink.tsx

```diff
@@ -1,26 +1,69 @@
 'use client';
 
-import { chakra } from '@chakra-ui/react';
+import { Text } from '@/ui/Text';
+import { Flex, Icon, chakra } from '@chakra-ui/react';
+import { ArrowRight } from '@phosphor-icons/react';
 import { ReactNode, forwardRef } from 'react';
 
-import { Button, ButtonProps } from './Button';
 import { Link, LinkProps } from './Link';
 import { NextLink } from './NextLink';
-import { buttonRecipe } from './theme/recipes/ButtonRecipe';
-
-const ButtonLinkBase = forwardRef<
-  HTMLAnchorElement,
-  { children: ReactNode; isExternal?: boolean; buttonProps?: ButtonProps; linkProps?: LinkProps }
->(({ children, isExternal, buttonProps, linkProps }, ref) =>
-  isExternal ? (
-    <Link variant="noUnderline" ref={ref} {...linkProps}>
-      <Button {...buttonProps}>{children}</Button>
-    </Link>
-  ) : (
-    <NextLink variant="noUnderline" ref={ref} {...linkProps}>
-      <Button {...buttonProps}>{children}</Button>
-    </NextLink>
-  )
+import { linkRecipe } from './theme/recipes/LinkRecipe';
+
+type ButtonLinkSize = 'big' | 'small';
+
+type ButtonLinkProps = Omit<LinkProps, 'variant' | 'size'> & {
+  children: ReactNode;
+  buttonLinkSize: ButtonLinkSize;
+};
+
+const ButtonLinkBase = forwardRef<HTMLAnchorElement, ButtonLinkProps>(
+  ({ children, buttonLinkSize, ...linkProps }, ref) => {
+    const isExternal = linkProps?.href?.startsWith('http');
+
+    const content = (
+      <Flex alignItems="center" gap={1} className="group">
+        <Text
+          textStyle={buttonLinkSize === 'big' ? 'text-medium-sm' : 'text-medium-xs'}
+          color="textPrimary"
+        >
+          {children}
+        </Text>
+        <Icon
+          w={buttonLinkSize === 'big' ? 3.5 : 3}
+          h={buttonLinkSize === 'big' ? 3.5 : 3}
```

Comment:
Can we stick to the size naming used in other components (md, lg, etc) instead of 'big'?
Can we use a recipe to style this component instead of the custom markup?